packages/web3/src/utils/error.ts

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+class CompilationError {
+  constructor(
+    public lineStart: number,
+    public column: number,
+    public errorType: string,
+    public line: number,
+    public codeLine: string,
+    public errorIndicator: string,
+    public message: string,
+    public additionalLine1?: string,
+    public additionalLine2?: string
+  ) {}
+
+  reformat(line: number, file: string): string {
+    const spaces = `${line}`.replace(/\d/g, ' ')
+    const newError = `${file} (${line}:${this.column}): ${this.errorType}
+${line} |${this.codeLine}
+${spaces} |${this.errorIndicator}
+${spaces} |${this.message}`
+
+    if (this.additionalLine1 && this.additionalLine2) {
```

Comment:
Q: What is `additionalLine1` and `additionalLine2`? Is it footer?

Reply:
Some errors can have a "hint" footer, see [this example from the test](https://github.com/alephium/alephium-web3/blob/1a9019ae2678ba60e9a81fdf49954966831e7892/packages/web3/src/utils/error.test.ts#L39-L44) Those are "real" error generated by our node `CompilerSpec`